src/components/ens-profile/ActionButtons/WatchButton.tsx

```diff
@@ -0,0 +1,138 @@
+import { toLower } from 'lodash';
+import React, { useCallback, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import GradientOutlineButton from '../GradientOutlineButton/GradientOutlineButton';
+import { useTheme } from '@rainbow-me/context';
+import { ColorModeProvider } from '@rainbow-me/design-system';
+import { removeWalletData } from '@rainbow-me/handlers/localstorage/removeWallet';
+import {
+  useAccountProfile,
+  useImportingWallet,
+  useInitializeWallet,
+  useWallets,
+} from '@rainbow-me/hooks';
+import {
+  addressSetSelected,
+  walletsSetSelected,
+  walletsUpdate,
+} from '@rainbow-me/redux/wallets';
+import { logger } from '@rainbow-me/utils';
+
+export default function WatchButton({
+  address: primaryAddress,
+  ensName,
+}: {
+  address?: string;
+  ensName?: string;
+}) {
+  const { colors } = useTheme();
+
+  const dispatch = useDispatch();
+
+  const { wallets } = useWallets();
+
+  const [watchingWalletId, watchingWallet] = useMemo(() => {
+    return (
+      Object.entries(wallets || {}).find(([_, wallet]: [string, any]) =>
+        wallet.addresses.some(({ address }: any) => address === primaryAddress)
+      ) || ['', '']
+    );
+  }, [primaryAddress, wallets]);
+  const isWatching = useMemo(() => Boolean(watchingWallet), [watchingWallet]);
+
+  const deleteWallet = useCallback(() => {
+    const newWallets = {
+      ...wallets,
+      [watchingWalletId]: {
+        ...wallets[watchingWalletId],
+        addresses: wallets[
+          watchingWalletId
+        ].addresses.map((account: { address: string }) =>
+          toLower(account.address) === toLower(primaryAddress)
+            ? { ...account, visible: false }
+            : account
+        ),
+      },
+    };
+    // If there are no visible wallets
+    // then delete the wallet
+    const visibleAddresses = newWallets[watchingWalletId].addresses.filter(
+      (account: { visible: boolean }) => account.visible
+    );
+    if (visibleAddresses.length === 0) {
+      // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+      delete newWallets[watchingWalletId];
+      dispatch(walletsUpdate(newWallets));
```

Comment:
i see this method in `ChangeWalletSheet` wondering if we should add it in a hook or redux/wallet to have the same code in one place